src/mame/machine/hpc3.cpp

```diff
@@ -150,7 +150,7 @@ void hpc3_device::device_start()
 		save_item(NAME(m_pbus_dma[i].m_config), i);
 		save_item(NAME(m_pbus_dma[i].m_control), i);
 
-		m_pbus_dma[i].m_timer = timer_alloc(TIMER_PBUS_DMA + i);
+		m_pbus_dma[i].m_timer = timer_alloc(FUNC(hpc3_device::do_pbus_dma), this);
```

Comment:
This is not quite correct: While `do_pbus_dma` is called, there is already a guard in `do_pbus_dma`, and only thing that it will do for channels >= 4 will be to stop the timer.